### PR TITLE
Netatmo o auth2 improvements v2

### PR DESCRIPTION
**Pull request by LoukKamp**
This PR replaces the PR with the same name without the addition of V2 and has the same changes, except for the extern trees

### CODE REVIEW

**Comment by gizmocuz** [comment]
Thank you for this PR!

Could you please have a look at the changes of hardware.html/js, there are 'way' too many changes here.
Like almost every line... that should not have been done.
All else is looking perfect to me

**Comment by LoukKamp** [comment]
Just rebuild my system from this branch and tested: All OK finally :)

This after reviewing all the changes in all the changed files and found no more unexpected differences.

So, as far as it concerns me; this PR is ready to merge.

File: hardware/Hardware.html
Comment (LoukKamp) [inline hardware/Hardware.html:1]: This file is misplaced (wrong directory) and will be deleted in following commits

File: hardware/Hardware.js
Comment (LoukKamp) [inline hardware/Hardware.js:1]: This file is misplaced (wrong directory) and will be deleted in following commits

**Comment by LoukKamp** [comment]
**Please hold** on for a moment: I found that it displays too much information when a user has selected the wrong scope and is not allowed to read data from a device.

**Comment by LoukKamp** [comment]
While working on improvement of exception messages, I came along this code:

```
                bRet = ParseJSon(sResult, root);
                if ((!bRet) || (!root.isObject()))
                {
                        Log(LOG_ERROR, "Invalid data received (GetMeterDetails)...");
                        return;
                }
                if (!root["error"].empty())
                {
                        // We received an error
                        Log(LOG_ERROR, "%s", root["error"]["message"].asString().c_str());
                        m_isLogged = false;
                        return;
                }
```
I used the log line _Log(LOG_ERROR, "%s", root["error"]["message"].asString().c_str());_ to display the error code in a similar situation, but that lead to a runtime exception that took about 10 pages of text. I assume this statement caused a stack corruption, because it tried to reference a 2-dimensional array with only 1 dimension. When I just removed the second index (["message"]), it worked as expected.

I'm not familiar with the exact description of the API, but I suspect that an exception on this level doesn't use a JSON message contains more than one level.

So, I have strong suspicions that the code above will crash the thread, Domiticz, or even the OS when it is executed. 

My suggestion is: to remove the second index or test the number of dimensions before referencing higher levels.

**Comment by LoukKamp** [comment]
Tested all kinds of exceptional situations, like wrong client_id, client_sectret, scope, disabled Netatmo app, no internet connection, and so on.

Again:  As far as it concerns me; this PR is ready to merge.

**Comment by gizmocuz** [comment]
Thank you very much for this PR and all the work!!!!! Much appreciated!!!!
I will make a new beta in the next hours

**Comment by LoukKamp** [comment]
Happy to be able to contribute, and thanks again for your help and patience! 

I tried to level up my own copy of this repository, but the only option I had was creating a pull request, so I did that, but that's not what I intended to do (now I have a new PR on this repository with one entry: the merge action). Not a big problem, because I will use this to upload a small typo in an alert message and the join of the Disable- and EnableUpdateAndDeleteButtons function.

I had more trouble updating my local repo on the PI (ended up with merge conflicts with my own uploaded code !???!), so I removed the repository on my PI and did a new clone. This with the consequence that I needed to rebuild the whole system.

However, while running the cmake command, I noticed the following:

```
-- Submodule update
Submodule 'extern/fmtlib' (https://github.com/fmtlib/fmt.git) registered for path 'extern/fmtlib'
Submodule 'extern/jsoncpp' (https://github.com/domoticz/jsoncpp) registered for path 'extern/jsoncpp'
Submodule 'minizip' (https://github.com/domoticz/minizip.git) registered for path 'extern/minizip'
Submodule 'extern/mosquitto' (https://github.com/eclipse/mosquitto.git) registered for path 'extern/mosquitto'
Submodule 'extern/sqlite-amalgamation' (https://github.com/azadkuh/sqlite-amalgamation.git) registered for path 'extern/sqlite-amalgamation'
Cloning into '/home/pi/myOwnDomoticz/extern/fmtlib'...
Cloning into '/home/pi/myOwnDomoticz/extern/jsoncpp'...
Cloning into '/home/pi/myOwnDomoticz/extern/minizip'...
Cloning into '/home/pi/myOwnDomoticz/extern/mosquitto'...
Cloning into '/home/pi/myOwnDomoticz/extern/sqlite-amalgamation'...
Submodule path 'extern/fmtlib': checked out '516a2e204921589824be0a7566be85c223e18ec6'
Submodule path 'extern/jsoncpp': checked out '2a6e163b02ac5044d74a44ed15f8aa7af687aea5'
Submodule path 'extern/minizip': checked out '78eb93e8e0ba228f37b197d2022d66b73856c9ff'
Submodule path 'extern/mosquitto': checked out '15292b20b0894ec7c5c3d47e4b22ee9d89f91132'
Submodule path 'extern/sqlite-amalgamation': checked out '15d0ff10ebc7e7225eced1de84bb52137000899b'

```
Right after this, the extern trees re-appeared in my commit commands:

```
root@DomoticzServer:/home/pi/myOwnDomoticz# git diff
diff --git a/extern/fmtlib b/extern/fmtlib
index a1c6bfd77..516a2e204 160000
--- a/extern/fmtlib
+++ b/extern/fmtlib
@@ -1 +1 @@
-Subproject commit a1c6bfd77b409f21e234cc7042795ca4d31fa020
+Subproject commit 516a2e204921589824be0a7566be85c223e18ec6
diff --git a/extern/jsoncpp b/extern/jsoncpp
index 3beb37ea1..2a6e163b0 160000
--- a/extern/jsoncpp
+++ b/extern/jsoncpp
@@ -1 +1 @@
-Subproject commit 3beb37ea14aec1bdce1a6d542dc464d00f4a6cec
+Subproject commit 2a6e163b02ac5044d74a44ed15f8aa7af687aea5
diff --git a/extern/minizip b/extern/minizip
index f52826430..78eb93e8e 160000
--- a/extern/minizip
+++ b/extern/minizip
@@ -1 +1 @@
-Subproject commit f5282643091dc1b33546bb8d8b3c23d78fdba231
+Subproject commit 78eb93e8e0ba228f37b197d2022d66b73856c9ff
diff --git a/extern/mosquitto b/extern/mosquitto
index 22e0a4ba9..15292b20b 160000
--- a/extern/mosquitto
+++ b/extern/mosquitto
@@ -1 +1 @@
-Subproject commit 22e0a4ba9bd6b0ce0f430638a54e6d9d1ca4af59
+Subproject commit 15292b20b0894ec7c5c3d47e4b22ee9d89f91132
diff --git a/extern/sqlite-amalgamation b/extern/sqlite-amalgamation
index cd9454659..15d0ff10e 160000
--- a/extern/sqlite-amalgamation
+++ b/extern/sqlite-amalgamation
@@ -1 +1 @@
-Subproject commit cd945465998165e6ec5dbb302cda29723927fe84
+Subproject commit 15d0ff10ebc7e7225eced1de84bb52137000899b

```

So, I wonder if the above logging has something related to these differences?

**Pull request merged**